kore/src/Kore/Attribute/Pattern/Simplified.hs

```diff
@@ -0,0 +1,154 @@
+
+instance Debug Simplified
+
+instance Diff Simplified where
+    diffPrec = diffPrecIgnore
+
+instance NFData Simplified
+
+instance Hashable Simplified
+
+alwaysSimplified :: a -> Simplified
+alwaysSimplified = const (Simplified True)
+{-# INLINE alwaysSimplified #-}
+
+notSimplified :: a -> Simplified
+notSimplified = const (Simplified False)
+{-# INLINE notSimplified #-}
+
+instance Synthetic Simplified (Bottom sort) where
+    synthetic = alwaysSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Top sort) where
+    synthetic = alwaysSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Const StringLiteral) where
+    synthetic = alwaysSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Const CharLiteral) where
+    synthetic = alwaysSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Const (UnifiedVariable variable)) where
+    synthetic = alwaysSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Exists sort variable) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Forall sort variable) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (And sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Or sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Not sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Application head) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Ceil sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Floor sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (DomainValue sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Equals sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (In sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Implies sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Iff sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Mu variable) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Nu variable) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Next sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Rewrites sort) where
+    synthetic = notSimplified
+    {-# INLINE synthetic #-}
+
+instance Synthetic Simplified (Builtin sort) where
```

Comment:
Shouldn't this be always true for ints, bools and strings, and shouldn't this be easy to compute for map, lists and sets?

Reply:
For Int, Bool, and String: yes, these are always simplified. It cannot be computed for Map, List, or Set: being "simplified" isn't a property of the pattern, it tells the history of the pattern, that is, "this pattern has been run through the simplified since the last time a substitution was applied."

Reply:
Ok, should this instance return "alwaysSimplified" for those cases?

Reply:
Yes, I think it should.